parsl/providers/lsf/lsf.py

```diff
@@ -93,8 +112,26 @@ def __init__(self,
                          launcher=launcher)
 
         self.project = project
+        self.queue = queue
+        self.cores_per_block = cores_per_block
+        self.cores_per_node = cores_per_node
         self.move_files = move_files
-        self.scheduler_options = scheduler_options
+        self.bsub_redirection = bsub_redirection
+        self.request_by_nodes = request_by_nodes
+
+        # Update scheduler options
+        self.scheduler_options = scheduler_options + "\n"
+        if project:
+            self.scheduler_options += "#BSUB -P {}\n".format(project)
+        if queue:
+            self.scheduler_options += "#BSUB -q {}\n".format(queue)
+        if request_by_nodes:
+            self.scheduler_options += "#BSUB -nnodes {}\n".format(nodes_per_block)
+        if not request_by_nodes and cores_per_block and cores_per_node:
```

Comment:
this `if` won't properly lead the user into setting cores_per_block/cores_per_node if they omit it - it will instead mean that neither an -nnodes or -n gets added.

A better structure might be:
if request by noes: 
     set -nnodes
else:
     assert cores_per_block not none, "useful error/hint"
     assert cores_per_node not none, "useful error/hint"
     add -n and -R

Reply:
Done. I have updated the `if` logic for `request_by_nodes`